pep-0484.txt

```diff
@@ -1063,6 +1063,54 @@ in turn, to ``collections.abc.Callable``::
       ...
 
 
+The ``NoReturn`` type
+---------------------
+
+The ``typing`` module provides a special type ``NoReturn`` to annotate functions
+that never return normally. For example, a function that unconditionally
+raises an exception::
+
+  from typing import NoReturn
+
+  def stop() -> NoReturn:
+      raise Exception('no way')
+
+Such return annotation is used in ``typeshed`` stubs for functions
+such as ``sys.exit``. Static type checkers will ensure that functions
+annotated as returning ``NoReturn`` truly never return, either implicitly
+or explicitly::
+
+  import sys
+  from typing import NoReturn
+
+    def f(x: int) -> NoReturn:  # Error, f(0) implicitly returns None
+        if x != 0:
+            sys.exit(1)
+
+The checkers will also recognize that the code after calls to such functions
+is unreachable and will behave accordingly::
+
+  # continue from first example
+  def g(x: int) -> int:
+      if x > 0:
+          return x
+      stop()
+      return 'whatever works'  # No error reported by checkers that
```

Comment:
I find this debatable. Why would you have dead code that doesn't type-check? (Though maybe we've had this discussion already? I don't feel strongly about it, just wonder if maybe PEP 484 isn't the place to require this.)